### ISSUE TITLE
SPIKE: Find a tool for running scenarios

### ISSUE THREAD

**Issue by ThetaSinner** [holochain/wind-tunnel#111]
The intended deployment pattern for Wind Tunnel was to colocate a binary for Holochain and the scenario on each test node and start many scenario+Holochain pairs across a network. This has the advantage of scaling better than the TryCP method we've landed on for now. It also removes the extra network traffic that we currently require to coordinate scenarios.

If we don't solve this problem, then we need to rewrite the remainder of the original wind tunnel scenarios that were meant to be run this way. They would have to be re-written to use TryCP instead.

AC:
- [x] Any tool we pick must be simple to deploy, operate and maintain.
- [x] The tool should be simple enough to learn that we can have multiple people learn to use it.
- [x] We don't generally require synchronization or communication between the scenario instances but we should be able to start them all at a similar time. I.e. It wouldn't be acceptable to start them in series if we have to wait for Holochain to start up with each one.
- [x] It must be possible, with documentation, to run the tool locally so that we can iterate on the setup.
- [x] Should support running on our supported platforms. That would be Mac, Linux, Windows, Android (see tx5 CI) and iOS. It doesn't have to work up front, we just want to be able to get there.
  - This is relevant to packaging. If we're using Nix to make the scenario binaries then we'd need cross compilation to support Windows. Maybe this is a good reason to consider using Rust for building scenarios?

**Comment by cdunster** [comment]
One solution that I've explored is just using Nix as all the machines are running NixOS. Therefore, we can build each scenario on the CI machine and then `nix copy` them to the test machines and then run them on those machines all via SSH. This just requires a couple of lines of bash and for the SSH permissions to be setup.

See here for an example: https://github.com/cdunster/nix-testing/blob/14d03a7db461f4d51c4888085e029422f40c81f5/flake.nix#L44-L48

**Comment by cdunster** [comment]
Another solution I explored for this was [Nomad](https://www.nomadproject.io), mentioned above. Nomad is designed for this kind of thing and works well with docker containers. It also supports running non-containerised applications, so that is what I tried.

The example can be found at: https://github.com/cdunster/nomad-example

Nomad seems ideal for what we want with the ability to specify how many instances we want for each task which would equate to the number of agents required for each test scenario, we can also build once (in CI?), upload the binary somewhere, and then download it as an artefact in the Nomad config to be ran in a task. Nomad also provides a nice dashboard and other features. This approach will allow us to move away from packaging the scenarios with Nix if we want to.
However, I did run into some issues getting the example just to run so this makes me concerned about other issues with larger and more complex examples, it's also another tool to learn and maintain support for.

I also need to explore the potential of generating telemetrics and other data and sending that back to the host/server to upload to the InfluxDB instance.

All in all, I think that either of these two final options (`nix copy` or Nomad) are viable but I would propose starting with Nomad and if we run into issues, it seems not suitable, or we get blocked, then we can discuss and maybe just create our own solution with `nix copy` and other nix commands.
